napari/tests/test_napari.py

```diff
@@ -1,46 +1,70 @@
+    (LayerTypes.IMAGE, np.random.random((10, 15)), 2, None),
+    (LayerTypes.VOLUME, np.random.random((10, 15, 20)), 3, None),
+    (LayerTypes.PYRAMID, [np.random.random(s) for s in [(40, 20), (20, 10), (10, 5)]], 2, None),
+    (LayerTypes.LABELS, np.random.randint(20, size=(10, 15)), 2, None),
+    (LayerTypes.POINTS, 20 * np.random.random((10, 2)), 2, None),
+    (LayerTypes.VECTORS, 20 * np.random.random((10, 2, 2)), 2, None),
+    (LayerTypes.SHAPES, 20 * np.random.random((10, 4, 2)), 2, None),
+    (LayerTypes.SURFACE, (np.random.random((10, 3)), np.random.randint(10, size=(6, 3)),
+                          np.random.random(10)), 3, 2),
+    (LayerTypes.SURFACE, (np.random.random((10, 3)), np.random.randint(10, size=(6, 3)),
+                          np.random.random(10)), 3, 3)
 
-    viewer = napari.view_image(data)
+])
+def test_view(qtbot, layer_type, data, ndim, ndisplay):
+
+    np.random.seed(0)
+    viewer = view_layer_type(layer_type, data, ndisplay)
     view = viewer.window.qt_viewer
     qtbot.addWidget(view)
-
+    viewer.dims.ndisplay = ndim
     assert np.all(viewer.layers[0].data == data)
-
     assert len(viewer.layers) == 1
     assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
 
-    assert viewer.dims.ndim == 2
+    assert viewer.dims.ndim == ndim
     assert view.dims.nsliders == viewer.dims.ndim
     assert np.sum(view.dims._displayed_sliders) == 0
 
     # Close the viewer
     viewer.window.close()
 
-    data = np.random.random((10, 15, 20))
-    viewer = napari.view_image(data)
-    view = viewer.window.qt_viewer
-    qtbot.addWidget(view)
-    viewer.dims.ndisplay = 3
-
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 3
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
 
-    # Close the viewer
-    viewer.window.close()
+def view_layer_type(layer_type, data, ndisplay):
+    """
+    Convenience method that maps a LayerType to it's view method.
+
+    Parameters
+    ----------
+    layer_type : LayerTypes
+        Layer type to view
+    data :
+        The layer data to view
+    """
+    if layer_type is LayerTypes.IMAGE:
+        return napari.view_image(data)
+    if layer_type is LayerTypes.VOLUME:
+        return napari.view_image(data)
+    if layer_type is LayerTypes.PYRAMID:
+        return napari.view_image(data, is_pyramid=True)
+    if layer_type is LayerTypes.LABELS:
+        return napari.view_labels(data)
+    if layer_type is LayerTypes.POINTS:
+        return napari.view_points(data)
+    if layer_type is LayerTypes.VECTORS:
+        return napari.view_vectors(data)
+    if layer_type is LayerTypes.SHAPES:
+        return napari.view_shapes(data)
+    if layer_type is LayerTypes.SURFACE:
+        return napari.view_surface(data, ndisplay=ndisplay)
```

Comment:
no need for special case ndisplay setting here - but we should have **kwargs on all the `view_*` methods